### PR TITLE
Introduce new REV_SERVER environment variables. Deprecate CONDITIONAL_FORWARDING ones, but maintain backward compatibility

### PR DESCRIPTION
**Pull request by PromoFaux**
## Description
Companion PR to be merged **_only after_** (and if) https://github.com/pi-hole/pi-hole/pull/3882 is accepted and merged /released

## Motivation and Context
The CONDITIONAL_FORWARDING variables have been replaced with REV_SERVER variables - this allows a docker user to specify them in their environment variables, whilst maintaining backward compatibility for the old style ones to be used (and transformed by the core repo)

## How Has This Been Tested?
I've not tested the docker image - honestly I'm not entirely sure how

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Review by diginc — APPROVED** [review]

File: start.sh [hunk 1]
```diff
@@ -61,10 +65,18 @@ change_setting "IPV6_ADDRESS" "$ServerIPv6"
 change_setting "DNS_BOGUS_PRIV" "$DNS_BOGUS_PRIV"
 change_setting "DNS_FQDN_REQUIRED" "$DNS_FQDN_REQUIRED"
 change_setting "DNSSEC" "$DNSSEC"
-change_setting "CONDITIONAL_FORWARDING" "$CONDITIONAL_FORWARDING"
-change_setting "CONDITIONAL_FORWARDING_IP" "$CONDITIONAL_FORWARDING_IP"
-change_setting "CONDITIONAL_FORWARDING_DOMAIN" "$CONDITIONAL_FORWARDING_DOMAIN"
-change_setting "CONDITIONAL_FORWARDING_REVERSE" "$CONDITIONAL_FORWARDING_REVERSE"
+change_setting "REV_SERVER"="$REV_SERVER"
```
Comment (diginc) [inline start.sh:68]: should the `=` be there?

Comment (PromoFaux) [reply start.sh:68]: Good catch, left over copy-paste mistake.

Comment (PromoFaux) [reply start.sh:68]:
```suggestion
change_setting "REV_SERVER" "$REV_SERVER"
```

**Review by michelle350 — COMMENTED** [review]
> ``

**Comment by PromoFaux** [comment]
To note: this also supersedes https://github.com/pi-hole/docker-pi-hole/pull/707

File: README.md [hunk 2]
```diff
@@ -123,6 +123,16 @@ There are other environment variables if you want to customize various things in
...
 
+## Deprecated environment variables:
+While these may still work, they are likely to be removed in a future version. Where applicible, alternative variable names are indicated
+
+| Docker Environment Var. | Description | Replaced By |
+| ----------------------- | ----------- | ----------- |
+| `CONDITIONAL_FORWARDING: <"true"\|"false">`<br/> *Optional* *Default: "false"* | Enable DNS conditional forwarding for device name resolution | `REV_SERVER`|
+| `CONDITIONAL_FORWARDING_IP: <Router's IP>`<br/> *Optional* | If conditional forwarding is enabled, set the IP of the local network router | `REV_SERVER_TARGET` |
+| `CONDITIONAL_FORWARDING_DOMAIN: <Network Domain>`<br/> *Optional* | If conditional forwarding is enabled, set the domain of the local network router | `REV_SERVER_DOMAIN` | 
+| `CONDITIONAL_FORWARDING_REVERSE: <Reverse DNS>`<br/> *Optional* | If conditional forwarding is enabled, set the reverse DNS of the local network router (e.g. `0.168.192.in-addr.arpa`) | `REV_SERVER_CIDR` |
```
Comment (DL6ER) [inline README.md:134]: We should add a warning that `REV_SERVER_CIDR` is **not** expecting said 

> (e.g. `0.168.192.in-addr.arpa`)

Comment (DL6ER) [reply README.md:134]: Yes, but that is 20 lines up. You know ... just trying to anticipate possible misunderstanding from those not carefully reading the guide but rather skimming over them.

**Review by DL6ER — APPROVED** [review]

**Comment by dschaper** [comment]
Sorry, Actions was jammed with cancelled events and apparently the only way to clear them is to delete that failed workflow and trigger with the same event again.

**Pull request merged**